app/models/reports/performance_score.rb

```diff
@@ -0,0 +1,49 @@
+module Reports
+  class PerformanceScore
+    CONTROL_SCORE_WEIGHT = 0.5
+    VISITS_SCORE_WEIGHT = 0.3
+    REGISTRATIONS_SCORE_WEIGHT = 0.2
+    TARGET_REGISTRATION_RATE = 0.1
+
+    def initialize(region:, result:)
+      @region = region
+      @result = result
+    end
+
+    def overall_score
+      # max score: 100
+      (control_score + visits_score + registrations_score)
+    end
+
+    def control_score
+      CONTROL_SCORE_WEIGHT * @result.controlled_patients_rate.values.last
+    end
+
+    def visits_score
+      VISITS_SCORE_WEIGHT * visits_rate
+    end
+
+    def visits_rate
+      100 - @result.missed_visits_rate.values.last
+    end
+
+    def registrations_score
+      REGISTRATIONS_SCORE_WEIGHT * registrations_rate
+    end
+
+    def registrations_rate
+      registrations = @result.registrations.values.last
+
+      # If the target is zero, return 100% if any registrations occurred
+      if target_registrations <= 0
```

Comment:
I'm hitting an error locally related to this part of the scoring:

```
[2020-10-02T14:56:28.514-05:00] FATAL: NoMethodError (undefined method `/' for nil:NilClass):

[2020-10-02T14:56:28.515-05:00] FATAL: app/models/reports/performance_score.rb:42:in `registrations_rate'
app/models/reports/performance_score.rb:31:in `registrations_score'
app/models/reports/performance_score.rb:15:in `overall_score'
app/controllers/my_facilities/ranked_facilities_controller.rb:41:in `block in show'
app/controllers/my_facilities/ranked_facilities_controller.rb:41:in `sort_by'
app/controllers/my_facilities/ranked_facilities_controller.rb:41:in `show'
app/controllers/my_facilities/ranked_facilities_controller.rb:65:in `block in set_time_zone'
app/controllers/my_facilities/ranked_facilities_controller.rb:65:in `set_time_zone'
app/controllers/admin_controller.rb:24:in `switch_locale'
```

I'm guessing you need to add a check for nil `registrations`.

Reply:
Added code to handle `nil` for all `Result` used here